plugins/org.eclipse.xtext.xtext.ui/src/org/eclipse/xtext/xtext/ui/wizard/project/AdvancedNewProjectPage.xtend

```diff
@@ -73,18 +87,115 @@ class AdvancedNewProjectPage extends WizardPage {
 				sourceLayout = DropDown[
 					enabled = true
 					items = SourceLayout.values.map[toString]
-				]
+				].decorate(INFORMATION, "Info about layout")
+
+			]
+
+			statusWidget = new StatusWidget(it, SWT.NONE) => [
+				layoutData = new GridData(SWT.FILL, SWT.BOTTOM, true, true)
 			]
 		]
 
-		createUiProject.require(createIdeProject)
-		createIdeaProject.require(createIdeProject)
-		createWebProject.require(createIdeProject)
-		makeUiProjectRequirePlainLayout
-		makeWebProjectRequireMavenOrGradle
+		val selectionControl = new SelectionListener() {
+			override widgetSelected(SelectionEvent e) {
+				validate(e)
+			}
+
+			override widgetDefaultSelected(SelectionEvent e) {
+				validate(e)
+			}
+		}
+		createUiProject.addSelectionListener(selectionControl)
+		sourceLayout.addSelectionListener(selectionControl)
+		createWebProject.addSelectionListener(selectionControl)
+		preferredBuildSystem.addSelectionListener(selectionControl)
+		createUiProject.addSelectionListener(selectionControl)
+		createIdeaProject.addSelectionListener(selectionControl)
+		createWebProject.addSelectionListener(selectionControl)
+		createIdeProject.addSelectionListener(selectionControl)
 		setDefaults
 	}
 
+	def void validate(SelectionEvent e) {
+		statusWidget.clearStatus
+		checkWidgets(e)
+		pageComplete = statusWidget.severtity !== ERROR
+	}
+
+	def checkWidgets(SelectionEvent e) {
+		val source = e?.source
+		if (createUiProject.selection && sourceLayout.selectionIndex != 0) {
+			if (createUiProject === source) {
+				reportIssue(ERROR, '''
+				'�createUiProject.text�' requier Flat source layout.
+				Please <a>select 'Flat'</a> source layout.''', [
```

Comment:
Ah, you offer "quick fixes". In that case, why don't you always offer both solutions?